apps/site/test/site_web/controllers/schedule/finder_api_test.exs

```diff
@@ -0,0 +1,332 @@
+defmodule SiteWeb.ScheduleController.FinderApiTest do
+  use SiteWeb.ConnCase
+
+  @moduletag :external
+
+  alias Predictions.Prediction
+  alias Routes.Route
+  alias Services.Repo, as: ServicesRepo
+  alias Schedules.{Schedule, Trip}
+  alias Stops.Stop
+
+  @now Util.now()
+  @schedule_time Timex.shift(@now, minutes: 3)
+  @prediction_time Timex.shift(@now, minutes: 5)
+
+  @stop %Stop{id: "place-sstat"}
+  @trip %Trip{direction_id: 0, id: "CR-Weekday-Fall-19-839"}
+  @route %Route{
+    id: "CR-Providence",
+    type: 2,
+    direction_destinations: %{0 => "Stoughton or Wickford Junction", 1 => "South Station"}
+  }
+
+  @schedule %Schedule{
+    route: @route,
+    trip: @trip,
+    stop: @stop,
+    time: @schedule_time,
+    flag?: false,
+    early_departure?: false,
+    last_stop?: true,
+    stop_sequence: 10,
+    pickup_type: 1
+  }
+
+  @prediction %Prediction{
+    departing?: true,
+    direction_id: 0,
+    id: "prediction-39783543-70050-60",
+    route: @route,
+    schedule_relationship: nil,
+    status: nil,
+    stop: @stop,
+    stop_sequence: 0,
+    time: @prediction_time,
+    track: "2",
+    trip: @trip
+  }
+
+  describe "journeys/2" do
+    test "gets valid journeys in order to derive trip params", %{conn: conn} do
+      route_id = "CR-Worcester"
+      date = get_valid_trip_date(route_id)
+      conn = assign(conn, :date, date)
+
+      journey =
+        %{id: route_id, direction: "0", stop: "place-WML-0199"}
+        |> get_valid_journeys(conn)
+        |> List.first()
+
+      assert %{"trip" => %{"id" => _}, "route" => %{"id" => _}} = journey
+      assert %{"departure" => %{"schedule" => %{"time" => date_time_string}}} = journey
+      assert {:ok, date_time, _} = DateTime.from_iso8601(date_time_string)
+      assert date == date_time |> DateTime.to_date() |> Date.to_iso8601()
+    end
+
+    test "successfully calls the API for bus", %{conn: conn} do
```

Comment:
We have enough current issues with tests relying on live data that I think we should strongly prefer not to introduce new ones. We've already built a way of dependency-injecting the API calls (`schedules_fn` and `predictions_fn`) which we're using elsewhere in these tests — can we use those in all the tests, and delete the ones that are just "it does the API call"?